src/core/components/push_payload.js

```diff
@@ -0,0 +1,537 @@
+  }
+
+  set title(value: ?string) {
+    if (!value || !value.length) return;
+
+    this._payload.aps.alert.title = value;
+    this._title = value;
+  }
+
+  get subtitle() {
+    return this._subtitle;
+  }
+
+  set subtitle(value: ?string) {
+    if (!value || !value.length) return;
+
+    this._payload.aps.alert.subtitle = value;
+    this._subtitle = value;
+  }
+
+  get body() {
+    return this._body;
+  }
+
+  set body(value: ?string) {
+    if (!value || !value.length) return;
+
+    this._payload.aps.alert.body = value;
+    this._body = value;
+  }
+
+  get badge() {
+    return this._badge;
+  }
+
+  set badge(value: ?number) {
+    if (value === undefined || value === null) return;
+
+    this._payload.aps.badge = value;
+    this._badge = value;
+  }
+
+  get sound() {
+    return this._sound;
+  }
+
+  set sound(value: ?string) {
+    if (!value || !value.length) return;
+
+    this._payload.aps.sound = value;
+    this._sound = value;
+  }
+
+  set silent(value: boolean) {
+    this._isSilent = value;
+  }
+
+  _setDefaultPayloadStructure() {
+    this._payload.aps = { alert: {} };
+  }
+
+  toObject(): ?Object {
+    let payload = Object.assign({}, this._payload);
+    /** @type {{alert: Object, badge: number, sound: string}} */
+    let aps = payload.aps;
+    let alert = aps.alert;
+
+    if (this._isSilent) {
+      aps['content-available'] = 1;
+    }
+
+    if (this._apnsPushType === 'apns2') {
+      if (!this._configurations || !this._configurations.length) {
+        throw new ReferenceError('APNS2 configuration is missing');
+      }
+
+      let configurations = [];
+      this._configurations.forEach((configuration: APNS2Configuration) => {
+        configurations.push(this._objectFromAPNS2Configuration(configuration));
+      });
+
+      if (configurations.length) {
+        payload.pn_push = configurations;
+      }
+    }
+
+    if (!alert || !Object.keys(alert).length) {
+      delete aps.alert;
+    }
+
+    if (this._isSilent) {
+      delete aps.alert;
+      delete aps.badge;
+      delete aps.sound;
+      alert = {};
+    }
+
+    return this._isSilent || Object.keys(alert).length ? payload : null;
+  }
+
+  _objectFromAPNS2Configuration(configuration: APNS2Configuration) {
+    if (!configuration.targets || !configuration.targets.length) {
+      throw new ReferenceError('At least one APNS2 target should be provided');
+    }
+
+    let targets = [];
+    configuration.targets.forEach((target: APNS2Target) => {
+      targets.push(this._objectFromAPNSTarget(target));
+    });
+
+    const { collapseId, date } = configuration;
+    let objectifiedConfiguration: {
+      auth_method: string,
+      targets: Array<Object>,
+      version: string,
+      collapse_id?: string,
+      expiration?: string
+    } = { auth_method: 'token', targets, version: 'v2' };
+
+    if (collapseId && collapseId.length) {
+      objectifiedConfiguration.collapse_id = collapseId;
+    }
+
+    if (date) {
```

Comment:
Is it ever possible to have a non-date object in the `configuration.date` value? To be save you can check: `if (date && typeof date.toISOString !== undefined) {`

Reply:
With Flow types we ask user pass Date there, but sure, user may pass whatever he want there.  This is actually reason, why we write docs - to help user figure out what exactly he did wrong (with this check we will silently ignore mistake - not really good pattern).